File: src/test/scala/com/github/tetrisanalyzer/move/MoveTest.scala
package com.github.tetrisanalyzer.move

import com.github.tetrisanalyzer.BaseTest
import org.junit.{Before, Test}
import rotation.AnticlockwiseRotation

class MoveTest extends BaseTest {
  var move: Move = _

  @Before def setUp {
    move = Move(0,5, 5)
  }

  @Test def rotate {
    move.rotate(new AnticlockwiseRotation, 1) should be (Move(1,5, 5))
  }

  @Test def left {
    move.left should be (Move(0,4, 5))
  }

  @Test def right {
    move.right should be (Move(0,6, 5))
  }

  @Test def up {
    move.up should be (Move(0,5, 4))
  }

  @Test def down {
    move.down should be (Move(0,5, 6))
  }

  @Test def testToString {
    move.toString should be ("(0,5, 5)")
  }
}

File: langs/java/src/test/java/com/github/tetrisanalyzer/move/MoveTest.java
package com.github.tetrisanalyzer.move;

import com.github.tetrisanalyzer.move.rotation.AnticlockwiseRotation;
import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class MoveTest {
    private Move move;

    @Before
    public void setUp() {
        move = new Move(0, 5, 5);
    }

    @Test
    public void rotate() {
        assertEquals(new Move(1, 5, 5), move.rotate(new AnticlockwiseRotation(), 1, 0, 0));
    }

    @Test
    public void left() {
        assertEquals(new Move(0, 4, 5), move.left());
    }

    @Test
    public void right() {
        assertEquals(new Move(0, 6, 5), move.right());
    }

    @Test
    public void up() {
        assertEquals(new Move(0, 5, 4), move.up());
    }

    @Test
    public void down() {
        assertEquals(new Move(0, 5,6), move.down());
    }

    @Test
    public void testToString() {
        assertEquals("(0,5, 5)", move.toString());
    }
}
